src/components/GestureComponents.tsx

```diff
@@ -1,63 +1,82 @@
-/* eslint-disable eslint-comments/no-unlimited-disable */
-/* eslint-disable */
-// @ts-nocheck TODO(TS) provide types
-import React from 'react';
-import ReactNative from 'react-native';
+import React, { PropsWithChildren } from 'react';
+import {
+  ScrollView as RNScrollView,
+  ScrollViewProps as RNScrollViewProps,
+  Switch as RNSwitch,
+  SwitchProps as RNSwitchProps,
+  TextInput as RNTextInput,
+  TextInputProps as RNTextInputProps,
+  DrawerLayoutAndroid as RNDrawerLayoutAndroid,
+  DrawerLayoutAndroidProps as RNDrawerLayoutAndroidProps,
+  FlatList as RNFlatList,
+  FlatListProps as RNFlatListProps,
+} from 'react-native';
 
 import createNativeWrapper from '../handlers/createNativeWrapper';
 
-const MEMOIZED = new WeakMap();
+import { NativeViewGestureHandlerProperties } from '../handlers/NativeViewGestureHandler';
 
-function memoizeWrap(Component, config) {
-  if (Component == null) {
-    return null;
-  }
-  let memoized = MEMOIZED.get(Component);
-  if (!memoized) {
-    memoized = createNativeWrapper(Component, config);
-    MEMOIZED.set(Component, memoized);
-  }
-  return memoized;
-}
+export const ScrollView = createNativeWrapper<
+  PropsWithChildren<RNScrollViewProps>
+>(RNScrollView, {
+  disallowInterruption: true,
+  shouldCancelWhenOutside: false,
+});
+// backward type compatibility with https://github.com/software-mansion/react-native-gesture-handler/blob/db78d3ca7d48e8ba57482d3fe9b0a15aa79d9932/react-native-gesture-handler.d.ts#L440-L457
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export type ScrollView = typeof ScrollView & {
+  scrollTo(
+    y?: number | { x?: number; y?: number; animated?: boolean },
+    x?: number,
+    animated?: boolean
+  ): void;
+  scrollToEnd(options?: { animated: boolean }): void;
+};
+export const Switch = createNativeWrapper<RNSwitchProps>(RNSwitch, {
```

Comment:
Wouldn't adding blank lines between components be good?

Reply:
Fixed in 62d66d6.